Sources/Interaction/Manipulators/MouseCameraAxisRotateManipulator/index.js

```diff
@@ -0,0 +1,140 @@
+import { vec3, mat4 } from 'gl-matrix';
+import macro from 'vtk.js/Sources/macro';
+import vtkCompositeCameraManipulator from 'vtk.js/Sources/Interaction/Manipulators/CompositeCameraManipulator';
+import vtkCompositeMouseManipulator from 'vtk.js/Sources/Interaction/Manipulators/CompositeMouseManipulator';
+import * as vtkMath from 'vtk.js/Sources/Common/Core/Math';
+
+// ----------------------------------------------------------------------------
+// vtkMouseCameraAxisRotateManipulator methods
+// ----------------------------------------------------------------------------
+
+function vtkMouseCameraAxisRotateManipulator(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkMouseCameraAxisRotateManipulator');
+
+  const newCamPos = new Float64Array(3);
+  const newFp = new Float64Array(3);
+  const newViewUp = new Float64Array(3);
+  const trans = new Float64Array(16);
+  const v2 = new Float64Array(3);
+  const centerNeg = new Float64Array(3);
+  const direction = new Float64Array(3);
+
+  publicAPI.onButtonDown = (interactor, renderer, position) => {
+    model.previousPosition = position;
+  };
+
+  publicAPI.onMouseMove = (interactor, renderer, position) => {
+    if (!position) {
+      return;
+    }
+
+    const camera = renderer.getActiveCamera();
+    const cameraPos = camera.getPosition();
+    const cameraFp = camera.getFocalPoint();
+
+    mat4.identity(trans);
+
+    const { center, rotationFactor } = model;
+
+    // Translate to center
+    mat4.translate(trans, trans, center);
+
+    const dx = model.previousPosition.x - position.x;
+    const dy = model.previousPosition.y - position.y;
+
+    const size = interactor.getView().getSize();
+
+    // Azimuth
+    const viewUp = camera.getViewUp();
+    mat4.rotate(
+      trans,
+      trans,
+      vtkMath.radiansFromDegrees(((360.0 * dx) / size[0]) * rotationFactor),
+      model.rotationAxis
+    );
+
+    // Elevation
+    vtkMath.cross(camera.getDirectionOfProjection(), viewUp, v2);
+    mat4.rotate(
+      trans,
+      trans,
+      vtkMath.radiansFromDegrees(((-360.0 * dy) / size[1]) * rotationFactor),
+      v2
+    );
+
+    // Translate back
+    centerNeg[0] = -center[0];
+    centerNeg[1] = -center[1];
+    centerNeg[2] = -center[2];
+    mat4.translate(trans, trans, centerNeg);
+
+    // Apply transformation to camera position, focal point, and view up
+    vec3.transformMat4(newCamPos, cameraPos, trans);
+    vec3.transformMat4(newFp, cameraFp, trans);
+
+    if (model.useHalfAxis) {
+      // Try to reset new camera postion so we don't go on
+      // the other half of the rotation axis...
+
+      // @Ken I tried some stuff but that didn't work
```

Comment:
@martinken  This is where I need your help...

Reply:
```
// what is the current distance from pos to fp
const fpdistance = vec3.distance(newCamPos, newCamFP);

// what is the current direction from the center of rotation
// to the camera
vonst fpdirection = vec3.create();
vec3.subtract(fpdirection, newCamPos, newCamFP);

// what is the current distance from pos to center of rotation
const distance = vec3.distance(newCamPos, center);

// what is the current direction from the center of rotation
// to the camera
vec3.subtract(direction, newCamPos, center);

// project the rotation axis onto the direction
// so we know how much below the half plane we are
const dotp = vec3.dot(model.rotationAxis, direction);

// adjust the new camera position to bring it up to the half plane
vec3.scaleAndAdd(newCamPos, newCamPos, rotationAxis, -dotp);

// the above step will change the distance which might feel odd
// so the next couple lines retore the distance to the center
vec3.scaleAndAdd(newCamPos, center, direction, distance);

// now recompute the newCamFP to be the same direction
// and distance as it was before
// NOTE: this may feel a bit odd so it might need to be changed
// basically it will move the FP as if you went below the plane
// while not actually letting it go below the plane
vec3.scaleAndAdd(newCamFP, newCamPos, fpdirection, -fpdistance);
```